build/commands/lib/config.js

```diff
@@ -520,7 +518,6 @@ Config.prototype.buildArgs = function () {
     if (this.targetEnvironment) {
       args.target_environment = this.targetEnvironment
     }
-    args.enable_dsyms = true
```

Comment:
> Just to clarify, does iOS still get dSYM's on non-PR builds? We need them for crash symbolication when submitting to the App Store/TestFlight

yes. We should have `isBraveReleaseBuild() == true` and `isOfficialBuild() == true` set in release builds (including iOS). This will make [this declaration](https://source.chromium.org/chromium/chromium/src/+/main:build/config/apple/symbols.gni;l=17?q=enable_dsyms&sq=&ss=chromium) enable dSYMs as intended in release builds.